src/deploy/osp_deployer/deployer.py

```diff
@@ -263,6 +285,32 @@ def deploy():
     sys.exit(ret_code)
 
 
+def validate_hugepage_params(hugepage_size):
+    if hugepage_size not in valid_hugepage_size_values:
+        raise ValueError(
+            "Invalid huge page size {}. Valid values are {}".format(
+                hugepage_size, valid_hugepage_size_values))
+
+
+def validate_numa_params(hostos_cpus):
+    if hostos_cpus not in valid_hostos_cpu_values:
+        raise ValueError(
+            "Invalid hostos_cpus value {} valid values are {}.".format(
+                hostos_cpus, valid_hostos_cpu_values))
+
+
+def dell_nfv_validations(hpg_enable, enable_numa, hugepage_size, hostos_cpus):
+    # Dell Nfv feature related Input validations
+    if hpg_enable:
+        validate_hugepage_params(hugepage_size)
+    else:
+        raise ValueError("hpg_size is invalid.")
+    if enable_numa:
+        validate_numa_params(hostos_cpus)
+    else:
+        raise ValueError("numa_hostos_cpu_count is invalid.")
+
+
 if __name__ == "__main__":
```

Comment:
Those validate functions are not necessary , the profile implementation allows you to specify valid values for a given setting in the json file - i beleive those are already in there for the cSP profile. ( if not, that s where they need  to go)

Reply:
Fixed
Validation added to Json